lib/resty/http.lua

```diff
@@ -776,11 +776,65 @@ function _M.request_uri(self, uri, params)
     if not params.path then params.path = path end
     if not params.query then params.query = query end
 
-    local c, err = self:connect(host, port)
+    -- See if we should use a proxy to make this request
+    local proxy_host, proxy_port;
+    local proxy_uri = self:get_proxy_uri(scheme, host)
+    if proxy_uri then
+        local parsed_proxy_uri, err = self:parse_uri(proxy_uri, false)
+        if not parsed_proxy_uri then
+            return nil, err
+        end
+
+        proxy_host, proxy_port = parsed_proxy_uri[2], parsed_proxy_uri[3]
+    end
+
+    local c, err = self:connect(proxy_host or host, proxy_port or port)
     if not c then
         return nil, err
     end
 
+    if proxy_uri and scheme == "https" then
```

Comment:
What about extracting handling of the proxy_uri into own function? 

Like

```
local c, err

if proxy_uri then
  c, err = self:connect_proxy(proxy_uri, host, port, params)  
else
  c, err = self:connect(host, port)
end
```

Not only that would split the code into logical segments, but also would allow use of that function as low-level API.